cloud/aws/providerconfig/types.go

```diff
@@ -14,22 +14,154 @@
+	// instance.
+	SecurityGroups []AWSResourceReference
+
+	// Subnet is a reference to the subnet to use for this instance
+	Subnet AWSResourceReference
+
+	// Placement indicates where to create the instance in AWS
+	Placement Placement
+
+	// LoadBalancerIDs is the IDs of the load balancers to which the new instance
+	// should be added once it's created.
+	LoadBalancerIDs []string
+}
+
+// AWSResourceReference is a reference to a specific AWS resource by ID, ARN, or filters
+type AWSResourceReference struct {
+	// ID of resource
+	ID *string
+
+	// ARN of resource
+	ARN *string
+
+	// Filters is a set of filters used to identify a resource
+	Filters []Filter
+}
+
+// Placement indicates where to create the instance in AWS
+type Placement struct {
+	// Region is the region to use to create the insntance
+	Region string
+
+	// AvailabilityZone is the availability zone of the instance
+	AvailabilityZone string
+}
+
+// Filter is a filter used to identify an AWS resource
+type Filter struct {
+	// Name of the filter
+	Name string
+
+	// Values includes one or more filter values
+	Values []string
+}
+
+// TagSpecification is the name/value pair for a tag
+type TagSpecification struct {
+	// Name of the tag
+	Name string
+
+	// Value of the tag
+	Value string
 }
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 type AWSClusterProviderConfig struct {
 	metav1.TypeMeta `json:",inline"`
 }
 
+// AWSMachineProviderStatus is the type that will be embedded in a Machine.Status.ProviderStatus field.
+// It containsk AWS-specific status information.
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 type AWSMachineProviderStatus struct {
 	metav1.TypeMeta `json:",inline"`
+
+	// InstanceID is the instance ID of the machine created in AWS
+	InstanceID *string
+
+	// InstanceState is the state of the AWS instance for this machine
+	InstanceState *string
+
+	// Conditions is a set of conditions associated with the Machine to indicate
+	// errors or other status
+	Conditions []AWSMachineProviderCondition
+}
+
+// AWSMachineProviderConditionType is a valid value for AWSMachineProviderCondition.Type
+type AWSMachineProviderConditionType string
+
+// Valid conditions for an AWS machine instance
+const (
+	// MachineCreated indicates whether the machine has been created or not. If not,
+	// it should include a reason and message for the failure.
+	MachineCreated AWSMachineProviderConditionType = "MachineCreated"
+)
+
+// AWSMachineProviderCondition is a condition in a AWSMachineProviderStatus
+type AWSMachineProviderCondition struct {
+	// Type is the type of the condition.
+	Type AWSMachineProviderConditionType
+	// Status is the status of the condition.
+	Status corev1.ConditionStatus
+	// LastProbeTime is the last time we probed the condition.
+	// +optional
+	LastProbeTime metav1.Time
+	// LastTransitionTime is the last time the condition transitioned from one status to another.
```

Comment:
Is this "from one _state_ to another"?

Reply:
It's from one status to the other ... referring to the Status field above... when it transitions from corev1.StatusTrue to corev1.StatusFalse or viceversa.